R/class.R

```diff
@@ -327,3 +328,16 @@ str.S7_object <- function(object, ..., nest.lev = 0) {
 S7_class <- function(object) {
   attr(object, "S7_class", exact = TRUE)
 }
+
+
+check_prop_names <- function(parent, properties, error_call = sys.call(-1L)) {
+  forbidden <- c("names", "dim", "dimnames", "class", "tsp",
+                 "comment", "row.names",
+                 if ("factor" %in% class_dispatch(parent)) "levels")
```

Comment:
What's the justification for `levels`? I know the others get special handling at the C-level, but I didn't know about `factor`.

Also add `...` to handle #442?

Reply:
Good point.  Very early in R's history,  factors *were* dealt with in C code in parts,  but I think this and `"levels"` in particular should no longer be specials.    {{but then I did not search R's C sources now}}

Reply:
Indeed, levels isn't handled specially at the C level: https://github.com/wch/r-source/blob/c06d2576f10fc82a4e06120df79f32eeaf044f99/src/main/attrib.c#L257-L270

I included it not to avoid special C handlers, but rather to acknowledge that S7 might not "own" the attribute or be responsible for managing it. However, that's probably unnecessarily restrictive. I'll remove it and update the forbidden names list to include only the attribute names with special C handlers.